hive-llap/llap.sh

```diff
@@ -0,0 +1,353 @@
+readonly YARN_MAX_CONTAINER_MEMORY=$(bdconfig get_property_value --configuration_file='/etc/hadoop/conf/yarn-site.xml' --name yarn.scheduler.maximum-allocation-mb)
+readonly ADDITIONAL_MASTER=$(/usr/share/google/get_metadata_value attributes/dataproc-master-additional)
+readonly HAS_SSD=$(/usr/share/google/get_metadata_value attributes/ssd)
+readonly NUM_LLAP_NODES=$(/usr/share/google/get_metadata_value attributes/num-llap-nodes)
+readonly HADOOP_CONF_DIR='/etc/hadoop/conf'
+readonly HIVE_CONF_DIR='/etc/hive/conf'
+readonly DEFAULT_INIT_ACTIONS_REPO="gs://dataproc-initialization-actions"
+readonly EXECUTOR_SIZE=$(/usr/share/google/get_metadata_value attributes/exec_size_mb || echo 4096)
+
+# user supplied location for file to ingest
+readonly INIT_ACTIONS_REPO="$(/usr/share/google/get_metadata_value attributes/init-actions-repo || echo ${DEFAULT_INIT_ACTIONS_REPO})"
+# directory files ingestied will reside
+readonly INIT_ACTIONS_DIR='/usr/lib/hive-llap'
+
+function pre_flight_checks(){
+    # check for bad configurations
+    if [[ "${NUM_LLAP_NODES}" -ge "${WORKER_NODE_COUNT}" ]]; then
+        echo "LLAP node count equals total worker count. There are no nodes to support Tez AM's. Please reduce LLAP instance count and re-deploy." && exit 1
+    fi
+}
+
+#modify yarn-site.xml buy adjusting the classpath
+function configure_yarn_site(){
+    echo "configure yarn-site.xml..."
+
+    local yarnappclasspath="$(bdconfig get_property_value --configuration_file='/etc/hadoop/conf/yarn-site.xml' --name yarn.application.classpath)"
+
+    ###append new paths to the yarn.application.classpath
+    bdconfig set_property \
+    --configuration_file "/etc/hadoop/conf/yarn-site.xml" \
+    --name "yarn.application.classpath" \
+    --value "${yarnappclasspath},/usr/local/share/google/dataproc/lib/*,/usr/lib/hadoop/*,/usr/lib/hadoop/lib/*,/usr/lib/hadoop-hdfs/*,/usr/lib/hadoop-hdfs/lib/*,/usr/lib/hadoop-yarn/*,/usr/lib/hadoop-yarn/lib/*,/usr/lib/tez/*,/usr/lib/tez/lib/*" \
+    --clobber
+
+    # Ensure that the max container memory 
+    if [[ "${NODE_MANAGER_MEMORY}" != "${YARN_MAX_CONTAINER_MEMORY}" ]]; then
+       echo "not configured properly..."
+    fi
+}
+
+function download_init_actions() {
+    # Download initialization actions locally. This will download the start_llap.sh file to the cluster for execution Check if metadata is supplied
+    echo "downalod init actions supplied as metadata..."
+    mkdir -p "${INIT_ACTIONS_DIR}"
+    gsutil cp "${INIT_ACTIONS_REPO}/hive-llap/start_llap.sh" "${INIT_ACTIONS_DIR}"
+    chmod 700 "${INIT_ACTIONS_DIR}/start_llap.sh"
+}
+
+### add configurations to hive-site for LLAP
+function configure_hive_site(){
+
+    # different configuration if HA
+    echo "configure hive-site.xml...."
+
+    bdconfig set_property \
+    --configuration_file "${HIVE_CONF_DIR}/hive-site.xml" \
+    --name 'hive.llap.daemon.service.hosts' --value '@llap0' \
+    --clobber
```

Comment:
add indentation to the lines after the first line of a command.

Reply:
fixed